### ISSUE TITLE
Compile go and zig before running, instead of "go run" and "zig run"

### ISSUE THREAD

**Issue by Arxcis** [Arxcis/adventofcode2020#30]
Because they get unfair performance numbers, by compiling and running in a single command:
![image](https://user-images.githubusercontent.com/3092618/101208918-2b44cc00-3673-11eb-9bb1-303b7143ddcb.png)

There is no way to time only the run-time, when we compile and run with the same command, so we should split it up, to be able to only measure run-time:
```sh
start=$(($(date +%s%N)/1000000))
cat $INPUT | zig run $SOLUTION -O ReleaseFast | diff - $OUTPUT
end=$(($(date +%s%N)/1000000))
```

**Comment by Arxcis** [comment]
fixed for golang in 1e826a0
![image](https://user-images.githubusercontent.com/3092618/101240055-0a20c180-36ed-11eb-8f8b-69b630129574.png)

**Comment by Arxcis** [comment]
So I fixed this issue with golang, but I did not understand how to do the same using the ZIG CLI. @Avokadoen do you have any pointers for me?
I tried:
```sh
$ zig build-exe --name out.zig day01.zig 
error: unable to find zig installation directory: FileNotFound
```
Should this work? What am I doing wrong?

**Comment by Stektpotet** [comment]
This would also be a possibility for python, using
`python -m py_compile day01.py`

However it seems to act weirdly depending on the used way of reading `stdin` when running the compiled script.
_I'll have a deeper look..._

for now, a viable option is to either add an optional parameter of the `./languages/python.sh`, or as a separate script `./languages/python_c.sh` to use in the solutions people add to `test.sh`

**Comment by Arxcis** [comment]
>  or as a separate script `./languages/python_c.sh` to use in the solutions people add to `test.sh`

This! :+1:  So you could opt in by choosing which script you are running. `language/python.sh` vs `language/python-compiled.sh`

**Comment by Arxcis** [comment]
@Avokadoen seemed to have fixed this issue for Zig in  #59 . Thank you very much :pray:  Closing... 

![image](https://user-images.githubusercontent.com/3092618/101294545-4997fc80-3818-11eb-8466-7d45e2811eaf.png)

**Comment by Arxcis** [comment]
Interestingly `.zig` seem to suffer from the same bottleneck as `.c` on day01. `.go`'s dominance continues.
